en/Sorting Algorithms/Recursive Versions/Recursive Bubble Sort.md

```diff
@@ -0,0 +1,72 @@
+# Recursive Bubble Sort
+
+Bubble Sort is one of the simplest sorting algorithms that compares two elements at a time, and swaps them if they are in the wrong order. This process is repeated until the entire sequence is in order.
+
+- Time Complexity: `O(n ^ 2)` for average case; `O(n)` for best case
+- Space Complexity: `O(n)`; note that iterative bubble sort has space complexity as `O(1)`
+
+## Steps
+
+Base case: If the size of the array is 1, return.
+
+- We need to fix the last element of the current sub-array. For this, iterate over the entire array using normal Bubble Sort, and perform swapping.
+- Next, call the function on the entire array excluding the last element(which was fixed by the iteration in the above step)
+- Repeat until Base Case is reached.
+
+## Example:
+
+Let the given array be: `{5, 3, 2, 1, 4}`
+
+__First Pass:__
```

Comment:
Shouldn't `pass` be `test`?

Reply:
Hrmmm. I think First `Iteration` would be more proper? By pass, I meant the number of times I have traversed the array.

Reply:
OK, then use `iteration` instead. 🙂

Reply:
Thanks! Fixed in latest commit